b64.h

```diff
@@ -28,50 +28,55 @@ char *b64Encode(char *data)
         'w', 'x', 'y', 'z', '0', '1', '2', '3',
         '4', '5', '6', '7', '8', '9', '+', '/'};
 
-    ssize_t inlen = strlen(data);
+    int external_flag = 0;
```

Comment:
if you do `char *dst` then all this `external_flag` stuff becomes `if (!ds) dst = malloc(outlen + 1);